Polyfills/Canvas/Source/Context.cpp

```diff
@@ -86,17 +86,18 @@ namespace Babylon::Polyfills::Internal
         JsRuntime::NativeObject::GetFromJavaScript(env).Set(JS_CONTEXT_CONSTRUCTOR_NAME, func);
     }
 
-    Napi::Value Context::CreateInstance(Napi::Env env, NativeCanvas* canvas)
+    Napi::Value Context::CreateInstance(Napi::Env env, Napi::Value canvas)
     {
         Napi::HandleScope scope{env};
 
         auto func = JsRuntime::NativeObject::GetFromJavaScript(env).Get(JS_CONTEXT_CONSTRUCTOR_NAME).As<Napi::Function>();
-        return func.New({Napi::External<NativeCanvas>::New(env, canvas)});
+        return func.New({canvas});
     }
 
     Context::Context(const Napi::CallbackInfo& info)
         : Napi::ObjectWrap<Context>{info}
-        , m_canvas{info[0].As<Napi::External<NativeCanvas>>().Data()}
+        , m_canvasObject{Napi::Weak(info[0].As<Napi::Object>())}
```

Comment:
Does there need to be a js reference from context to canvas? If the js side only references the context, we don't want the canvas to be gc'd, right? I assume this is how it works in the browser?

Reply:
The reference from context to canvas is needed by the `getCanvas` method.
>  If the js side only references the context, we don't want the canvas to be gc'd, right? I assume this is how it works in the browser?
That was my guess. But in that case, when is the Canvas and Context are collected if both keep a strong reference on each other? Is there a way to detect that?